pkg/distributor/distributor.go

```diff
@@ -1029,6 +1029,9 @@ func (d *Distributor) MetricsForLabelMatchers(ctx context.Context, from, through
 			}
 			ms := ingester_client.FromMetricsForLabelMatchersResponse(resp)
 			for _, m := range ms {
+				if err := queryLimiter.AddDataBytes(resp.Size()); err != nil {
```

Comment:
Why do we increase response size per matcher? I might miss something, but why we don't add response size once since it is a single request

Reply:
It should be outside the loop. Thanks for catching it. :)